Sources/MediaNavigationBar.swift

```diff
@@ -0,0 +1,135 @@
+/*****************************************************************************
+ * MediaNavigationBar.swift
+ *
+ * Copyright © 2019 VLC authors and VideoLAN
+ *
+ * Authors: Robert Gordon <robwaynegordon@gmail.com>
+ *
+ *
+ * Refer to the COPYING file of the official project for license.
+ *****************************************************************************/
+
+@objc (VLCMediaNavigationBarDelegate)
+protocol VLCMediaNavigationBarDelegate {
+    func didMinimizePlayback(_ navigationBar :MediaNavigationBar)
+    func didToggleChromeCast(_ navigationBar :MediaNavigationBar)
+    func didToggleAirPlay(_ navigationBar :MediaNavigationBar)
+}
+
+@objc (VLCMediaNavigationBar)
+@objcMembers class MediaNavigationBar :UIView {
+    
+    // MARK: Instance Variables
+    weak var delegate :VLCMediaNavigationBarDelegate?
+    
+    lazy var minimizePlaybackButton :UIButton = {
+        var minButton = UIButton(type: .system)
+        minButton.addTarget(self, action: #selector(minimizePlayback), for: .touchUpInside)
+        minButton.setImage(UIImage(named: "iconChevron"), for: .normal)
+        minButton.tintColor = .white
+        minButton.translatesAutoresizingMaskIntoConstraints = false
+        return minButton
+    }()
+    
+    private var mediaTitleTextLabel :UILabel = {
+        var label = UILabel()
+        label.numberOfLines = 1
+        label.textColor = .white
+        label.font = UIFont(name: "SFProDisplay-Medium", size: 17)
+        label.text = NSLocalizedString("TITLE", comment: "Video Title")
+        label.isUserInteractionEnabled = false
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+    
+    lazy var chromeCastButton :UIButton = {
+        var chromeButton = UIButton(type: .system)
+        chromeButton.addTarget(self, action: #selector(toggleChromeCast), for: .touchUpInside)
+        chromeButton.setImage(UIImage(named: "renderer"), for: .normal)
+        chromeButton.tintColor = .white
+        chromeButton.translatesAutoresizingMaskIntoConstraints = false
+        return chromeButton
+    }()
+    
+    lazy var airPlayButton :UIButton = {
+        var airPlayButton = UIButton(type: .system)
+        airPlayButton.addTarget(self, action: #selector(toggleAirPlay), for: .touchUpInside)
+        airPlayButton.setImage(UIImage(named: "iconAirplay"), for: .normal)
+        airPlayButton.tintColor = .white
+        airPlayButton.translatesAutoresizingMaskIntoConstraints = false
+        return airPlayButton
+    }()
+    
+    private func setupConstraints() {
+        NSLayoutConstraint.activate([
+            minimizePlaybackButton.leadingAnchor.constraint(equalTo: self.leadingAnchor),
+            minimizePlaybackButton.topAnchor.constraint(equalTo: self.topAnchor),
```

Comment:
no self

Reply:
this was more of a preference to use self for inherited properties but I'll remove it

Reply:
okay in general you want to avoid it because then you know in swift code, everywhere where you have self it's potential for a retain cycle because only there you get forced to use self